addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -246,7 +257,63 @@ export class LinkPlugin extends Plugin {
      */
     removeLink() {
         const link = this.linkElement;
-        unwrapContents(link);
+        const selection = this.shared.getEditableSelection();
+        if (link && link.isContentEditable) {
+            unwrapContents(link);
+        }
+        if (selection.anchorNode.isConnected) {
+            this.shared.setSelection(selection, { normalize: false });
+        } else {
+            this.shared.setSelection(this.shared.getEditableSelection());
+        }
+    }
+
+    removeLinkFromSelection() {
+        this.shared.splitSelection();
+        const selection = this.shared.getEditableSelection();
+
+        // If not, unlink only the part(s) of the link(s) that are selected:
+        // `<a>a[b</a>c<a>d</a>e<a>f]g</a>` => `<a>a</a>[bcdef]<a>g</a>`.
+        let { anchorNode, focusNode, anchorOffset, focusOffset } = selection;
+        const direction = getCursorDirection(anchorNode, anchorOffset, focusNode, focusOffset);
+        // Split the links around the selection.
+        const [startLink, endLink] = [
+            closestElement(anchorNode, "a"),
+            closestElement(focusNode, "a"),
+        ];
+        if (startLink) {
+            anchorNode = this.shared.splitAroundUntil(anchorNode, startLink);
+            anchorOffset = direction === DIRECTIONS.RIGHT ? 0 : nodeSize(anchorNode);
```

Comment:
why not use startContainer and startOffset (you don t need to care abount direction